ENG-12713 : Add micro-benchmarks

gernest opened the pull request:
This adds  benchmarks  to account for time/space efficiency of the the library implementation. The aim here is to cover all major performance hotspots and lay out foundation for detecting future regressions

nshi commented on voltdbclient/benchmarks_test.go:
```diff
@@ -0,0 +1,225 @@
+		}
+		if res == nil {
+
+		}
```
Unused code block.

nshi commented on voltdbclient/node_conn.go:
```diff
@@ -244,6 +244,23 @@ func (nc *nodeConn) loop(writer io.Writer, piCh <-chan *procedureInvocation, res
+				// 	[]byte(fmt.Sprint(handle)), 0600)
+				// if err != nil {
+				// 	log.Fatal(err)
+				// }
```
Does it make sense to put this in the test code instead of the driver itself or is the response object already out of context outside of this class?

gernest replied:
The API makes it hard to access the raw response  bytes. This is the only way I can access them at the moment.

Also, this is insignificant. The samples once created they don't have to be recreated again. I left them just in case I need to recreate the samples for some reasons I don't know yet.

gernest commented:
__TIFU__: After I had added the last change, I did a rebase on master. Fixed conflicts and pushed the changes, only a few mins later to realize what I have done.

This should be a heck of a PR. My git-fu aren't helping either. Any advice is welcome.

gernest commented:
Resubmitted again with reviewable changes #33`
